### ISSUE TITLE
Global_Prefs_Override.xml will do preferences higher than 100[pct]

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#771]
**Reported by Ageless on 8 May 38937968 07:33 UTC**
Something that came up through the Seti forums. An interesting bug, which seems to apply to all BOINC 6 versions.

When setting the preferences through the Local Advanced Preferences menu and setting the amount of CPUs to use to something higher than 100%, this setting is saved in the global_prefs_override.xml file. This bug works on all preferences where percentages are used.

Here's the excerpt of the global_prefs_override.xml file for the defect:
```
<global_preferences>
   <max_ncpus_pct>7542.000000</max_ncpus_pct>
   <ram_max_used_busy_pct>7523.000000</ram_max_used_busy_pct>
   <ram_max_used_idle_pct>9532.000000</ram_max_used_idle_pct>
   <vm_max_used_pct>546332.000000</vm_max_used_pct>
   <cpu_usage_limit>9832.000000</cpu_usage_limit>
</global_preferences>
```

Check the memory percentages I set. This translates in BOINC to:
```
08-Dec-08 02:29:34||Preferences limit memory usage when active to 154013.15MB
08-Dec-08 02:29:34||Preferences limit memory usage when idle to 195142.01MB
```

Cool... I only have 2GB and a 32bit OS... :-)

When I go back into the preferences menu, the percentages for the CPUs and throttle have reset, but for the memory and virtual memory settings the percentages stay at those high numbers.

I am not sure if it affects BOINC badly in any way, but a check is in order that resets the percentage written to the file to 100 if the number is over 100.

**Comment by romw** [comment]
**Commented by Ageless on 10 Oct 38937971 07:33 UTC**
Hmm, best set CC to me and assign to David. :-)

**Comment by romw** [comment]
**Commented by Ageless on 29 Nov 38937974 03:33 UTC**
typo.

**Comment by romw** [comment]
**Commented by davea on 11 Mar 38938293 03:33 UTC**
(In [- manager: clamp percentages at [0,100](16645])) in prefs dialog
    fixes #794
- wrapper: if app checkpoints, write CPU time to checkpoint file
    fixes #792 (I hope)
